lib/engine/game/base.rb

```diff
@@ -896,11 +896,17 @@ def sell_shares_and_change_price(bundle, allow_president_change: true, swap: nil
       end
 
       def log_share_price(entity, from)
-        to = entity.share_price.price
-        return unless from != to
+        msg = entity.share_price.report_last_move
```

Comment:
The stock market moves multiple times at the end of the stock round in some games (e.g. 1828). Make sure that case is handled properly.